includes/class-wc-checkout.php

```diff
@@ -1,156 +1,253 @@
+			do_action( 'woocommerce_checkout_init', self::$instance );
+		}
+		return self::$instance;
+	}
 
 	/**
-	 * @var WC_Checkout The single instance of the class
-	 * @since 2.1
+	 * See if variable is set. Used to support legacy public variables which are no longer defined.
+	 *
+	 * @param string $key
+	 * @return bool
 	 */
-	protected static $_instance = null;
-
-	/** @var Bool */
-	public $enable_guest_checkout;
+	public function __isset( $key ) {
+		return in_array( $key, array(
+			'enable_signup',
+			'enable_guest_checkout',
+			'must_create_account',
+			'checkout_fields',
+			'posted',
+			'shipping_method',
+			'payment_method',
+			'customer_id',
+			'shipping_methods',
+		) );
+	}
 
 	/**
-	 * Main WC_Checkout Instance.
+	 * Sets the legacy public variables for backwards compatibility.
 	 *
-	 * Ensures only one instance of WC_Checkout is loaded or can be loaded.
+	 * @param string $key
+	 * @param mixed $value
+	 */
+	public function __set( $key, $value ) {
+		switch ( $key ) {
+			case 'enable_signup' :
+				$bool_value = wc_string_to_bool( $value );
+
+				if ( $bool_value !== $this->is_registration_enabled() ) {
+					remove_filter( 'woocommerce_checkout_registration_enabled', '__return_true', 0 );
+					remove_filter( 'woocommerce_checkout_registration_enabled', '__return_false', 0 );
+					add_filter( 'woocommerce_checkout_registration_enabled', $bool_value ? '__return_true' : '__return_false', 0 );
+				}
+				break;
+			case 'enable_guest_checkout' :
+				$bool_value = wc_string_to_bool( $value );
+
+				if ( $bool_value !== $this->is_registration_required() ) {
+					remove_filter( 'woocommerce_checkout_registration_required', '__return_true', 0 );
+					remove_filter( 'woocommerce_checkout_registration_required', '__return_false', 0 );
+					add_filter( 'woocommerce_checkout_registration_required', $bool_value ? '__return_false' : '__return_true', 0 );
+				}
+				break;
+			case 'checkout_fields' :
+				$this->fields = $value;
+				break;
+			case 'shipping_methods' :
+				WC()->session->set( 'chosen_shipping_methods', $value );
+				break;
+			case 'legacy_posted_data' :
+				$this->legacy_posted_data = $value;
+				break;
+		}
+	}
+
+	/**
+	 * Gets the legacy public variables for backwards compatibility.
 	 *
-	 * @since 2.1
-	 * @static
-	 * @return WC_Checkout Main instance
+	 * @param string $key
+	 * @return string
 	 */
-	public static function instance() {
-		if ( is_null( self::$_instance ) )
-			self::$_instance = new self();
-		return self::$_instance;
+	public function __get( $key ) {
+		switch ( $key ) {
+			case 'enable_signup' :
+				return $this->is_registration_enabled();
+			case 'enable_guest_checkout' :
+				return ! $this->is_registration_required();
+			case 'must_create_account' :
+				return $this->is_registration_required() && ! is_user_logged_in();
+			case 'checkout_fields' :
+				return $this->get_checkout_fields();
+			case 'posted' :
+				if ( ! $this->legacy_posted_data ) {
```

Comment:
Since `get_posted_data` is a private function for this class - can we go ahead and use `get_posted_data` to do this check and just `return $this->get_posted_data();`? It would save a few lines in a few places.